pkg/datastore/example_multitenancy_test.go

```diff
@@ -0,0 +1,105 @@
+package datastore_test
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"math/rand"
+	"time"
+
+	"github.com/vmware-labs/multi-tenant-persistence-for-saas/pkg/authorizer"
+	"github.com/vmware-labs/multi-tenant-persistence-for-saas/pkg/datastore"
+	"github.com/vmware-labs/multi-tenant-persistence-for-saas/pkg/dbrole"
+)
+
+type User struct {
+	Id    string `gorm:"primaryKey"`
+	Name  string
+	Age   int
+	OrgId string `gorm:"primaryKey"`
+}
+
+func (p User) String() string {
+	return fmt.Sprintf("[%s] %s: %d", p.OrgId, p.Name, p.Age)
+}
+
+// Example for the multi-tenancy feature shows by creating records for different orgs with same IDs
+// and then checking that only the correct org context can access data belonging to their own org.
+func ExampleDataStore_multiTenancy() {
+	rand.Seed(time.Now().UnixNano())
+	uId := fmt.Sprintf("P%d", rand.Intn(1_000_0000))
+	p1 := &User{uId, "Bob", 31, "Coke"}
+	p2 := &User{uId, "John", 36, "Pepsi"}
+	p3 := &User{"P3", "Pat", 39, "Coke"}
+
+	TENANT_ADMIN := "tenant_admin"
+	TENANT_AUDITOR := "tenant_auditor"
+	mdAuthorizer := authorizer.MetadataBasedAuthorizer{}
+	CokeOrgCtx := mdAuthorizer.GetAuthContext("Coke", TENANT_ADMIN)
+	PepsiOrgCtx := mdAuthorizer.GetAuthContext("Pepsi", TENANT_ADMIN)
+
+	// Initializes the Datastore using the metadata authorizer and connection details obtained from the ENV variables.
+	ds, _ := datastore.FromEnv(datastore.GetCompLogger(), mdAuthorizer, nil)
```

Comment:
Same comment as for the multi-instance example - we shouldn't ignore an error here